pilot/pkg/config/kube/crd/types.gen.go

```diff
@@ -25,203 +25,273 @@ import (
 			},
 		},
-		Collection: &VirtualServiceList{},
+		Collection: &MeshPolicyList{},
 	},
-	schemas.Gateway.Type: {
-		Schema: schemas.Gateway,
-		Object: &Gateway{
+	collections.IstioAuthenticationV1Alpha1Policies.Resource().Kind(): {
+		Schema: collections.IstioAuthenticationV1Alpha1Policies,
+		Object: &Policy{
 			TypeMeta: meta_v1.TypeMeta{
-				Kind:       "Gateway",
-				APIVersion: APIVersion(&schemas.Gateway),
+				Kind:       "Policy",
+				APIVersion: APIVersion(collections.IstioAuthenticationV1Alpha1Policies),
 			},
 		},
-		Collection: &GatewayList{},
+		Collection: &PolicyList{},
 	},
-	schemas.ServiceEntry.Type: {
-		Schema: schemas.ServiceEntry,
-		Object: &ServiceEntry{
+	collections.IstioConfigV1Alpha2Adapters.Resource().Kind(): {
+		Schema: collections.IstioConfigV1Alpha2Adapters,
+		Object: &adapter{
 			TypeMeta: meta_v1.TypeMeta{
-				Kind:       "ServiceEntry",
-				APIVersion: APIVersion(&schemas.ServiceEntry),
+				Kind:       "adapter",
+				APIVersion: APIVersion(collections.IstioConfigV1Alpha2Adapters),
 			},
 		},
-		Collection: &ServiceEntryList{},
+		Collection: &adapterList{},
+	},
+	collections.IstioConfigV1Alpha2Httpapispecbindings.Resource().Kind(): {
+		Schema: collections.IstioConfigV1Alpha2Httpapispecbindings,
+		Object: &HTTPAPISpecBinding{
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       "HTTPAPISpecBinding",
+				APIVersion: APIVersion(collections.IstioConfigV1Alpha2Httpapispecbindings),
+			},
+		},
+		Collection: &HTTPAPISpecBindingList{},
+	},
+	collections.IstioConfigV1Alpha2Httpapispecs.Resource().Kind(): {
+		Schema: collections.IstioConfigV1Alpha2Httpapispecs,
+		Object: &HTTPAPISpec{
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       "HTTPAPISpec",
+				APIVersion: APIVersion(collections.IstioConfigV1Alpha2Httpapispecs),
+			},
+		},
+		Collection: &HTTPAPISpecList{},
+	},
+	collections.IstioConfigV1Alpha2Templates.Resource().Kind(): {
+		Schema: collections.IstioConfigV1Alpha2Templates,
+		Object: &template{
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       "template",
+				APIVersion: APIVersion(collections.IstioConfigV1Alpha2Templates),
+			},
+		},
+		Collection: &templateList{},
+	},
+	collections.IstioMeshV1Alpha1MeshConfig.Resource().Kind(): {
+		Schema: collections.IstioMeshV1Alpha1MeshConfig,
+		Object: &MeshConfig{
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       "MeshConfig",
+				APIVersion: APIVersion(collections.IstioMeshV1Alpha1MeshConfig),
+			},
+		},
+		Collection: &MeshConfigList{},
+	},
+	collections.IstioMixerV1ConfigClientQuotaspecbindings.Resource().Kind(): {
+		Schema: collections.IstioMixerV1ConfigClientQuotaspecbindings,
+		Object: &QuotaSpecBinding{
+			TypeMeta: meta_v1.TypeMeta{
+				Kind:       "QuotaSpecBinding",
+				APIVersion: APIVersion(collections.IstioMixerV1ConfigClientQuotaspecbindings),
+			},
+		},
+		Collection: &QuotaSpecBindingList{},
+	},
+	collections.IstioMixerV1ConfigClientQuotaspecs.Resource().Kind(): {
```

Comment:
can be done in a followup, we shouldn't make this by Kind since its not distinct

Reply:
ack